src/containers/PointOfCareApp.jsx

```diff
@@ -211,21 +232,52 @@ export class PointOfCareApp extends Component {
     }
 
     sourceActionIsDisabled = (element) => {
-        if (element.source) {
-            return false;
+        if (!element.source || element.source.sourceMessage === "") {
+            return true;
         }
-        return true;
+        return false;
     }
 
     nameSourceAction = (element) => {
         if (element.source) {
-            return (element.source instanceof Reference ? "Open Source Note" : "View Source");
+            return (element.source.note ? "Open Source Note" : (element.source.link ? "View Source Attachment" : (element.source.sourceMessage !== "" ? "View Source" : "No Source information")));
         }
         return "No source information";
     }
 
-    handleSnackbarClose = () => {
-        this.setState({ snackbarOpen: false });
+    getNoteModalStyle = () => {
```

Comment:
The styling of the noteModal shouldn't be dependent on whether or not the modal is being displayed; the styling should be constant, but the display of the modal should only ever happen when there is a note to display. Could we remove this conditional flag, or explain what bug was motivating it?